File: package/umt_python/tests/unit/string/test_snake_case.py
import unittest

from src.string import snake_case


class TestSnakeCase(unittest.TestCase):
    def test_camel_case(self):
        self.assertEqual(snake_case("helloWorld"), "hello_world")

    def test_space_separated(self):
        self.assertEqual(snake_case("Hello World"), "hello_world")

    def test_kebab_case(self):
        self.assertEqual(snake_case("foo-bar-baz"), "foo_bar_baz")

    def test_pascal_case(self):
        self.assertEqual(snake_case("FooBar"), "foo_bar")

    def test_empty_string(self):
        self.assertEqual(snake_case(""), "")

    def test_acronyms(self):
        self.assertEqual(snake_case("XMLHttpRequest"), "xml_http_request")


if __name__ == "__main__":
    unittest.main()

File: package/umt_rust/tests/string/test_snake_case.rs
use umt_rust::string::umt_snake_case;

#[test]
fn test_converts_camel_case() {
    assert_eq!(umt_snake_case("helloWorld"), "hello_world");
}

#[test]
fn test_converts_space_separated_words() {
    assert_eq!(umt_snake_case("Hello World"), "hello_world");
}

#[test]
fn test_converts_kebab_case() {
    assert_eq!(umt_snake_case("foo-bar-baz"), "foo_bar_baz");
}

#[test]
fn test_converts_pascal_case() {
    assert_eq!(umt_snake_case("FooBar"), "foo_bar");
}

#[test]
fn test_returns_empty_for_empty_string() {
    assert_eq!(umt_snake_case(""), "");
}

#[test]
fn test_handles_acronyms() {
    assert_eq!(umt_snake_case("XMLHttpRequest"), "xml_http_request");
}
